### PR TITLE
Update FAQ to clarify release model

### PR DESCRIPTION
**Pull request by hsaliak**
Minor updates to the gRPC IO FAQ.
- We clarify that we are part of CNCF.
- Updated discussion on roadmap to point to proposal repo
- Discussed release model. 

Please review the wording on the release model carefully.

### CODE REVIEW

File: faq/index.html [hunk 1]
```diff
@@ -45,11 +45,12 @@ <h6 class="arrow-r">Why would I want to use gRPC?</h6>
                     <li class="submenu">
                         <h6 class="arrow-r">Who’s using this and why?</h6>
                         <div class="submenu-content">
+                            <p> gRPC is a CNCF project.</p>
```
Comment (ejona86) [inline faq/index.html]: Link to CNCF's webpage?

File: faq/index.html [hunk 2]
```diff
@@ -87,10 +88,27 @@ <h6 class="arrow-r">Where is the documentation?</h6>
                     <li class="submenu">
                         <h6 class="arrow-r">What is the roadmap?</h6>
                         <div class="submenu-content">
-                            <p>The roadmap is being tracked in GitHub <a target="_blank" href="{{ site.data.config.milestones_link }}">milestones</a>. Please refer to the respective milestones in the gRPC, gRPC-java and gRPC-go repositories under the GitHub gRPC org.</p>
+                            <p>The gRPC project has an RFC process, through which new features are designed and approved for implementation. They are tracked in <a target="_blank" href="https://github.com/grpc/proposal"> this repository</a>. </p>
```
Comment (ejona86) [inline faq/index.html]: s#this repository#grpc/proposal#

File: faq/index.html [hunk 3]
```diff
@@ -87,10 +88,27 @@ <h6 class="arrow-r">Where is the documentation?</h6>
...
+                        </div>
+                    </li>
+                    <li class="submenu">
+                        <h6 class="arrow-r">What is the latest gRPC Version?</h6>
+                        <div class="submenu-content">
+                            <p>The latest release branch is {{site.data.config.grpc_release_branch}}</p>
```
Comment (ejona86) [inline faq/index.html]: Mention that the different repos may be slightly out-of-sync around the time of a release?

**Comment by jtattermusch** [comment]
please ignore: testing that CLA bot works.

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @hsaliak isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by hsaliak** [comment]
I've signed the CLA

**Comment by hsaliak** [comment]
I've signed the CLA.

**Pull request merged**